### PR TITLE
RFC: Storing users' team memberships

### PR DESCRIPTION
**Pull request by pivotal-jwinters**
[Proposal](https://github.com/pivotal-jwinters/rfcs/blob/proposal/group-storage/06-group-storage/proposal.md)

Status: Draft

This is a brain dump from a conversation we had about storing users' team memberships. This needs to be fleshed out a lot more, but it gives an early look at some of the options we've considered.

### CODE REVIEW

**Comment by Rukenshia** [comment]
@pivotal-jwinters Will this still synchronize with the login provider or would that mean that I would have to assign memberships within concourse? That would be a huge pain.

**Comment by pivotal-jwinters** [comment]
@Rukenshia there's a few different ways we could approach this, but I don't think much would change from the operators perspective.

File: 06-group-storage/proposal.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+- It's operationally less complex. We could probably get rid of the skymarshal component altogether and rely directly on `dex` tokens.
+
+
+### Disadvantages
+
+- It's slower. Every api request requires a database lookup.
+
+- It's a more complex data model. More things to manage in the database.
+
+- We still don't have full control over upstream groups. This means that if you're added to a new team in github, for example, you would still need to log out and log back in for us to pick up those changes.
```
Comment (pivotal-jamie-klassen) [inline 06-group-storage/proposal.md]: Not sure this should count as a Disadvantage of storing team memberships in the database, as this problem exists for both strategies.

Comment (pivotal-jwinters) [reply 06-group-storage/proposal.md]: good point, i removed it

**Comment by vito** [comment]
For anyone following along, this led to https://github.com/concourse/concourse/pull/4950 which has been merged and will be shipped in v6.1. Going to close this as there's not much point having an RFC for it retroactively.

**Pull request closed without merging**